### PR TITLE
add creation cspr modification network

### PR DESCRIPTION
**Pull request by ghazwarhili**

### CODE REVIEW

File: src/main/java/org/gridsuite/modification/server/modifications/ModificationUtils.java [hunk 1]
```diff
@@ -1126,6 +1110,48 @@ public void checkReactiveLimitsCreation(ReactiveLimitsHolderInfos modificationIn
...
+
+    public void checkStandbyAutomatonCreation(Boolean standby, Double b0, Double q0,
+                                              Double minSusceptance, Double maxSusceptance,
+                                              Double minQAtNominalV, Double maxQAtNominalV,
+                                              StaticVarCompensator.RegulationMode regulationMode,
+                                              NetworkModificationException.Type errorType,
+                                              String equipmentId,
+                                              String equipmentName) {
+        if (Boolean.TRUE.equals(standby) && regulationMode != StaticVarCompensator.RegulationMode.VOLTAGE) {
+            throw makeEquipmentException(errorType, equipmentId, equipmentName, "Regulation mode is expected to be voltage");
```
Comment (thangqp) [inline src/main/java/org/gridsuite/modification/server/modifications/ModificationUtils.java:1146]:
```suggestion
        if (Boolean.TRUE.equals(standby) && regulationMode != StaticVarCompensator.RegulationMode.VOLTAGE) {
            throw makeEquipmentException(errorType, equipmentId, equipmentName, "Standby is only supported only Voltage Regulation mode");
```

File: src/main/java/org/gridsuite/modification/server/modifications/ModificationUtils.java [hunk 2]
```diff
@@ -1126,6 +1110,48 @@ public void checkReactiveLimitsCreation(ReactiveLimitsHolderInfos modificationIn
...
+        }
+        if (Objects.nonNull(b0) && Objects.nonNull(minSusceptance) && Objects.nonNull(maxSusceptance) && (b0 < minSusceptance || b0 > maxSusceptance)
+            || Objects.nonNull(q0) && Objects.nonNull(minQAtNominalV) && Objects.nonNull(maxQAtNominalV) && (q0 < minQAtNominalV || q0 > maxQAtNominalV)) {
+            throw makeEquipmentException(errorType, equipmentId, equipmentName,
+                     "b0 is expected to be within the range of minimun susceptance and maximum susceptance");
```
Comment (thangqp) [inline src/main/java/org/gridsuite/modification/server/modifications/ModificationUtils.java:1151]:
```suggestion
            throw makeEquipmentException(errorType, equipmentId, equipmentName,
                     "b0 must be within the range of minimun susceptance and maximum susceptance");
```

File: src/test/java/org/gridsuite/modification/server/utils/ModificationCreation.java [hunk 3]
```diff
@@ -157,6 +157,18 @@ public static LoadModificationInfos getModificationLoad(String loadId, String vl
         return builder.build();
     }
 
+    public static StaticVarCompensatorCreationInfos getCreationStaticVarCompensator(String vlId, String loadId, String loadName, String busOrBusBarSectionId) {
```
Comment (thangqp) [inline src/test/java/org/gridsuite/modification/server/utils/ModificationCreation.java:160]: not used, to clean

File: src/main/java/org/gridsuite/modification/server/modifications/ModificationUtils.java [hunk 4]
```diff
@@ -1378,5 +1425,53 @@ public static void insertReportNode(ReportNode parent, ReportNode child) {
...
+                        ? injectionCreationInfos.getConnectionName()
+                        : injectionCreationInfos.getEquipmentId())
+                .withInjectionPositionOrder(position)
+                .withInjectionAdder(injectionAdder)
+                .build();
+        algo.apply(network, true, subReportNode);
+    }
+
+    public static void reportInjectionCreationConnectivity(InjectionCreationInfos injectionCreationInfos, ReportNode subReporter) {
+        if (injectionCreationInfos.getVoltageLevelId() == null || injectionCreationInfos.getBusOrBusbarSectionId() == null) {
```
Comment (EtienneLt) [inline src/main/java/org/gridsuite/modification/server/modifications/ModificationUtils.java:1446]: should it be better if merged with next if ?

File: src/main/java/org/gridsuite/modification/server/modifications/StaticVarCompensatorCreation.java [hunk 5]
```diff
@@ -0,0 +1,244 @@
...
+                .setBmin(staticVarCompensatorCreationInfos.getMinSusceptance())
+                .setVoltageSetpoint(staticVarCompensatorCreationInfos.getVoltageSetpoint())
+                .setReactivePowerSetpoint(staticVarCompensatorCreationInfos.getReactivePowerSetpoint())
+                .setRegulationMode(staticVarCompensatorCreationInfos.getRegulationMode())
+                .add();
+
+        addExtensionsToStaticVarCompensator(staticVarCompensatorCreationInfos, staticVarCompensator, voltageLevel, subReportNode);
+    }
+
+    private void createCompensatorVoltageRegulation(StaticVarCompensatorCreationInfos staticVarCompensatorCreationInfos,
```
Comment (EtienneLt) [inline src/main/java/org/gridsuite/modification/server/modifications/StaticVarCompensatorCreation.java:208]: createStaticVarCompensatorVoltageRegulation ?

**Review by EtienneLt — CHANGES_REQUESTED** [review]
small things

File: src/main/java/org/gridsuite/modification/server/modifications/ModificationUtils.java [hunk 6]
```diff
@@ -279,8 +288,7 @@ public void controlVoltageLevelCreation(VoltageLevelCreationInfos voltageLevelCr
         if (Objects.nonNull(voltageLevelCreationInfos.getIpMin()) && Objects.isNull(voltageLevelCreationInfos.getIpMax())) {
             throw new NetworkModificationException(CREATE_VOLTAGE_LEVEL_ERROR, "IpMax is required");
         }
-        if (Objects.nonNull(voltageLevelCreationInfos.getIpMin()) && Objects.nonNull(voltageLevelCreationInfos.getIpMax())
-            && voltageLevelCreationInfos.getIpMin() > voltageLevelCreationInfos.getIpMax()) {
+        if (Objects.nonNull(voltageLevelCreationInfos.getIpMin()) && voltageLevelCreationInfos.getIpMin() > voltageLevelCreationInfos.getIpMax()) {
```
[the remark this answers is not in the document]
Comment (ghazwarhili) [reply src/main/java/org/gridsuite/modification/server/modifications/ModificationUtils.java:291]: this code is not concerned by the aim of this PR 
I fixed because it is highlighted by sonar like warning

File: src/main/java/org/gridsuite/modification/server/modifications/ModificationUtils.java [hunk 7]
```diff
@@ -914,17 +922,11 @@ public <T> void reportElementaryCreation(ReportNode subReportNode, T value, Stri
...
-                return "    Fixed tap";
-            case CURRENT_LIMITER :
-                return "    Current limiter";
-            case ACTIVE_POWER_CONTROL :
-                return "    Active power control";
-            default :
-                return "";
-
-        }
+        return switch (regulationMode) {
```
[the remark this answers is not in the document]
Comment (ghazwarhili) [reply src/main/java/org/gridsuite/modification/server/modifications/ModificationUtils.java:925]: the same here, it is proposed by sonarLint 
"Switch statement can be replaced with enhanced 'switch' "

File: src/main/java/org/gridsuite/modification/server/dto/StaticVarCompensatorCreationInfos.java [hunk 8]
```diff
@@ -0,0 +1,109 @@
...
+    private Double reactivePowerSetpoint;
+
+    @Schema(description = "Voltage Regulation type")
+    private VoltageRegulationType voltageRegulationType;
+
+    @Schema(description = "Regulating terminal equipment id")
+    private String regulatingTerminalId;
+
+    @Schema(description = "Regulating terminal equipment type")
+    private String regulatingTerminalType;
```
Comment (EtienneLt) [inline src/main/java/org/gridsuite/modification/server/dto/StaticVarCompensatorCreationInfos.java:65]: maybe something like regulatingTerminalEquipmentType would me more readable ?

File: src/main/java/org/gridsuite/modification/server/entities/equipment/creation/StaticCompensatorCreationEntity.java [hunk 9]
```diff
@@ -0,0 +1,168 @@
...
+
+    @Enumerated(EnumType.STRING)
+    @Column(name = "voltageRegulationType")
+    private VoltageRegulationType voltageRegulationType;
+
+    @Column(name = "regulatingTerminalId")
+    private String regulatingTerminalId;
+
+    @Column(name = "regulatingTerminalType")
+    private String regulatingTerminalType;
```
Comment (EtienneLt) [inline src/main/java/org/gridsuite/modification/server/entities/equipment/creation/StaticCompensatorCreationEntity.java:58]: same here

File: src/main/java/org/gridsuite/modification/server/dto/StaticVarCompensatorCreationInfos.java (hunk 8 above)
Comment (ghazwarhili) [reply src/main/java/org/gridsuite/modification/server/dto/StaticVarCompensatorCreationInfos.java:65]: I used the same to be coherent with GeneratorCreation and TapChangerCreation

File: src/main/java/org/gridsuite/modification/server/entities/equipment/creation/StaticCompensatorCreationEntity.java (hunk 9 above)
Comment (ghazwarhili) [reply src/main/java/org/gridsuite/modification/server/entities/equipment/creation/StaticCompensatorCreationEntity.java:58]: I used the same to be coherent with GeneratorCreation and TapChangerCreation

**Review by EtienneLt — APPROVED** [review]

**Review by thangqp — APPROVED** [review]

File: src/main/java/org/gridsuite/modification/server/modifications/ModificationUtils.java [hunk 10]
```diff
@@ -1218,10 +1220,10 @@ public void checkMaxQGreaterThanMinQ(
     }
 
     public void checkMaxReactivePowerGreaterThanMinReactivePower(MinMaxReactiveLimits minMaxReactiveLimits, AttributeModification<Double> minimumReactivePowerInfo, AttributeModification<Double> maximumReactivePowerInfo, NetworkModificationException.Type exceptionType, String errorMessage) {
-        Double previousMinimumReactivePower = minMaxReactiveLimits.getMinQ();
-        Double previousMaximumReactivePower = minMaxReactiveLimits.getMaxQ();
-        Double minReactivePower = minimumReactivePowerInfo != null ? minimumReactivePowerInfo.getValue() : previousMinimumReactivePower;
-        Double maxReactivePower = maximumReactivePowerInfo != null ? maximumReactivePowerInfo.getValue() : previousMaximumReactivePower;
+        double previousMinimumReactivePower = minMaxReactiveLimits.getMinQ();
```
Comment (etiennehomer) [inline src/main/java/org/gridsuite/modification/server/modifications/ModificationUtils.java:1223]: Why this change ?

File: src/main/java/org/gridsuite/modification/server/modifications/GeneratorCreation.java [hunk 11]
```diff
@@ -241,38 +219,6 @@ private void updateGeneratorRegulatingTerminal(GeneratorCreationInfos generatorC
         }
     }
 
-    private void reportGeneratorConnectivity(GeneratorCreationInfos generatorCreationInfos, ReportNode subReporter) {
```
Comment (etiennehomer) [inline src/main/java/org/gridsuite/modification/server/modifications/GeneratorCreation.java:244]: It's better to make another PR previously for this type of change

File: src/main/java/org/gridsuite/modification/server/modifications/ModificationUtils.java [hunk 12]
```diff
@@ -1292,6 +1294,51 @@ public void checkReactiveLimitsCreation(ReactiveLimitsHolderInfos modificationIn
...
+        // check set points
+        if (Objects.requireNonNull(regulationMode) == StaticVarCompensator.RegulationMode.VOLTAGE && voltageSetpoint == null) {
+            throw makeEquipmentException(errorType, equipmentId, equipmentName, "Voltage setpoint is not set");
+        }
+        if (regulationMode == StaticVarCompensator.RegulationMode.REACTIVE_POWER && reactivePowerSetpoint == null) {
+            throw makeEquipmentException(errorType, equipmentId, equipmentName, "Reactive power setpoint is not set");
+        }
+    }
+
+    public void checkStandbyAutomatonCreation(Boolean standby, Double b0, Double q0,
```
Comment (etiennehomer) [inline src/main/java/org/gridsuite/modification/server/modifications/ModificationUtils.java:1325]: Give modificationInfos as parameter ?

File: src/main/java/org/gridsuite/modification/server/modifications/ModificationUtils.java (hunk 10 above)
Comment (ghazwarhili) [reply src/main/java/org/gridsuite/modification/server/modifications/ModificationUtils.java:1223]: it is highlighted by sonarLint to change the type to primitive type

File: src/main/java/org/gridsuite/modification/server/modifications/ModificationUtils.java [hunk 13]
```diff
@@ -1292,6 +1294,51 @@ public void checkReactiveLimitsCreation(ReactiveLimitsHolderInfos modificationIn
         }
     }
 
+    public void checkReactivePowerLimitsAndSetPointsCreation(Double maxSusceptance, Double minSusceptance,
```
[the remark this answers is not in the document]
Comment (ghazwarhili) [reply src/main/java/org/gridsuite/modification/server/modifications/ModificationUtils.java:1297]: ok

File: src/main/java/org/gridsuite/modification/server/modifications/ModificationUtils.java (hunk 12 above)
Comment (ghazwarhili) [reply src/main/java/org/gridsuite/modification/server/modifications/ModificationUtils.java:1325]: ok

**Review by etiennehomer — APPROVED** [review]
